js/main.js

```diff
@@ -23,17 +23,64 @@ function submited(event) {
   var dataObject = {
     titleText: title,
     notesText: notes,
-    imgUrl: img,
-    entryId: data.nextEntryId
+    imgUrl: img
   };
-  data.nextEntryId++;
-  data.entries.unshift(dataObject);
+  if (data.editing !== null) {
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === data.editing.entryId) {
+        data.entries[i].titleText = dataObject.titleText;
+        data.entries[i].notesText = dataObject.notesText;
+        data.entries[i].imgUrl = dataObject.imgUrl;
+        var $allList = document.querySelectorAll('li');
+        for (var j = 0; j < $allList.length; j++) {
+          if ($allList[j].getAttribute('data-entry-id') === data.editing.entryId.toString()) {
+            var $currentListItem = $allList[j];
+            $currentListItem.setAttribute('data-entry-id', data.editing.entryId);
+            var $createRow = document.createElement('div');
+            $createRow.setAttribute('class', 'row');
+            var $createColumn = document.createElement('div');
+            $createColumn.setAttribute('class', 'column-half');
+            var $createImg = document.createElement('img');
+            $createImg.setAttribute('src', dataObject.imgUrl);
+            $createImg.setAttribute('class', 'adjust-img column-full');
+            var $create2Column = document.createElement('div');
+            $create2Column.setAttribute('class', 'column-half');
+            var $createTitlediv = document.createElement('div');
+            $createTitlediv.setAttribute('class', 'list-title row adjust-position');
+            $createTitlediv.textContent = dataObject.titleText;
+            var $createIcon = document.createElement('i');
+            $createIcon.setAttribute('class', 'fa-solid fa-pen-to-square');
+            $createIcon.setAttribute('data-entry-id', dataObject.entryId);
+            var $createContentdiv = document.createElement('div');
+            $createContentdiv.setAttribute('class', 'list-content');
+            $createContentdiv.textContent = dataObject.notesText;
+
+            $createRow.appendChild($createColumn);
+            $createColumn.appendChild($createImg);
+            $createRow.appendChild($create2Column);
+            $create2Column.appendChild($createTitlediv);
+            $createTitlediv.appendChild($createIcon);
+            $create2Column.appendChild($createContentdiv);
+            $currentListItem.replaceWith($createRow);
```

Comment:
Don't you already have a function that generates a single DOM tree for a single Entry?